src/rcamera.h

```diff
@@ -134,434 +148,401 @@ void SetCameraMoveControls(int keyFront, int keyBack,
+// Note: angle must be provided in radians
+void CameraRoll(Camera3D *camera, float angle)
+{
+    // Rotation axis
+    Vector3 forward = GetCameraForward(camera);
 
-// Set camera alt key to combine with mouse movement (free camera)
-void SetCameraAltControl(int keyAlt) { CAMERA.altControl = keyAlt; }
+    // Rotate up direction around forward axis
+    camera->up = Vector3RotateAxis(camera->up, forward, angle);
+}
 
-// Set camera smooth zoom key to combine with mouse (free camera)
-void SetCameraSmoothZoomControl(int szoomKey) { CAMERA.smoothZoomControl = szoomKey; }
+// Moves camera slightly to simulate a walking motion
+// Note: Only active if camera->swingCounter > 0
+void CameraViewBobbing(Camera3D* camera)
+{
+    if (camera->swingCounter > 0)
+    {
+        // NOTE: We delay the target movement relative to the position movement to create a little pitch with each step.
+        camera->position.y = camera->position.y - 0.25f * sinf((camera->swingCounter + 1) / CAMERA_FIRST_PERSON_STEP_TRIGONOMETRIC_DIVIDER) / CAMERA_FIRST_PERSON_STEP_DIVIDER;
+        camera->target.y = camera->target.y - 0.25f * sinf((camera->swingCounter - 1) / CAMERA_FIRST_PERSON_STEP_TRIGONOMETRIC_DIVIDER) / CAMERA_FIRST_PERSON_STEP_DIVIDER;
+
+        // Update counter for next frame
+        camera->swingCounter %= 2147483647 /* INT_MAX */; // Counter must be positive
+        camera->swingCounter++;
+    }
+}
+
+// Returns the camera view matrix
+Matrix GetCameraViewMatrix(Camera3D *camera)
+{
+    return MatrixLookAt(camera->position, camera->target, camera->up);
+}
+
+// Returns the camera projection matrix
+Matrix GetCameraProjectionMatrix(Camera3D *camera, float aspect)
+{
+    if (camera->projection == CAMERA_PERSPECTIVE)
+    {
+        return MatrixPerspective(camera->fovy * DEG2RAD, aspect, CAMERA_CULL_DISTANCE_NEAR, CAMERA_CULL_DISTANCE_FAR);
+    }
+    else if (camera->projection == CAMERA_ORTHOGRAPHIC)
+    {
+        double top = camera->fovy / 2.0;
+        double right = top * aspect;
+        return MatrixOrtho(-right, right, -top, top, CAMERA_CULL_DISTANCE_NEAR, CAMERA_CULL_DISTANCE_FAR);
+    }
+    
+    return MatrixIdentity();
+}
 
-// Set camera move controls (1st person and 3rd person cameras)
-void SetCameraMoveControls(int keyFront, int keyBack, int keyRight, int keyLeft, int keyUp, int keyDown)
+
+#ifndef CAMERA_STANDALONE
+
+static int init_frames = 3; // TODO review and remove
+void UpdateCamera(Camera3D *camera)
 {
-    CAMERA.moveControl[MOVE_FRONT] = keyFront;
-    CAMERA.moveControl[MOVE_BACK] = keyBack;
-    CAMERA.moveControl[MOVE_RIGHT] = keyRight;
-    CAMERA.moveControl[MOVE_LEFT] = keyLeft;
-    CAMERA.moveControl[MOVE_UP] = keyUp;
-    CAMERA.moveControl[MOVE_DOWN] = keyDown;
+    // Avoid inital mouse "jump"
+    if (init_frames > 0) {
+        SetMousePosition(0, 0);
+        init_frames--;
+        return;
+    }
+    Vector2 mousePositionDelta = GetMouseDelta();
```

Comment:
Input states can be read and stored at the beginning of the function to avoid spreading the raylib-dependand functions all around the code.

Reply:
Makes sense to me - I'll change this!